### PR TITLE
feat: caching language header as part of key

### PR DESCRIPTION
**Pull request by TobiTenno**
### What did you fix? 

- fixes having to use the query param to cache on cloudflare/etc
- introduce more into settings, use it more heavily
- reorganize invocations to port/host, use dynamically
---

### Reproduction steps

---

### Considerations
- Does this contain a new dependency? **[No]**
- Does this introduce opinionated data formatting or manual data entry? **[No]**
- Does this pr include updated data files in a separate commit that can be reverted for a clean code-only pr? **[No]**
- Have I run the linter? **[No]**
- Is is a bug fix, feature request, or enhancement? **[Enhancement]**

### CODE REVIEW

File: src/controllers/index.js [hunk 1]
```diff
@@ -29,6 +29,8 @@ router.use((req, res, next) => {
   /* istanbul ignore if */
   if (!platforms.includes(req.platform)) req.platform = 'pc';
 
+  if (req.header('Accept-Language') && req.header('Accept-Language').length) res?.set('Vary', 'Accept-Language');
```
Comment (AyAyEm) [inline src/controllers/index.js:32]: That's not really a good looking solution to check for ` req.header('Accept-Language').length > 0`

**Review by AyAyEm — CHANGES_REQUESTED** [review]
Looks good, just add somewhere that you also refactored a bunch of stuff and also the reason if possible, it can be that you just felt like it.

File: src/controllers/index.js (hunk 1 above)
Comment (TobiTenno) [reply src/controllers/index.js:32]: i have to make sure the header exists first. this project (presently) doesn't have existential chaining coalescing at present, or i wouldn't have needed the first clause and could do `?.length`... and this is far more standard compared to `length > 0`, given that 0 is falsey already, so it's extraneous

**Comment by TobiTenno** [comment]
> Looks good, just add somewhere that you also refactored a bunch of stuff and also the reason, it can be that you just felt like it.

of course

File: src/controllers/index.js (hunk 1 above)
Comment (AyAyEm) [reply src/controllers/index.js:32]: well the problem is that it relies on implicit type conversion basically it's harder to read, but my fault as i did not include the full line
```suggestion
  if (req.header('Accept-Language') && req.header('Accept-Language').length > 0) res?.set('Vary', 'Accept-Language');
```

Comment (AyAyEm) [reply src/controllers/index.js:32]: In my view implicit type conversion in js is evil, but if you find it more readable and understandable the way you have already done it then no harm in it.

Comment (AyAyEm) [reply src/controllers/index.js:32]: this might be just a rant about me not liking js type system (ts is no better in this case)

Comment (TobiTenno) [reply src/controllers/index.js:32]: yeah, i'm not doing this, cause this is how ts would want you to do it anyway

Comment (TobiTenno) [reply src/controllers/index.js:32]: i was wrong anyway, we definitely have nullish coalescing available, and if i added `> 0`, it'd still be type converting

Comment (AyAyEm) [reply src/controllers/index.js]: yeah but it would be explicit

**Review by AyAyEm — APPROVED** [review]

**Pull request merged**